authentication-pre-request.js

```diff
@@ -0,0 +1,40 @@
+const { Header } = require('postman-collection');
+
+const { collectionVariables } = pm;
+
+const oauth2Request = {
+    url: 'https://api.oregonstate.edu/oauth2/token',
+    method: 'POST',
+    header: 'Content-Type:application/x-www-form-urlencoded',
+    body:`grant_type=client_credentials&scope=
+      &client_id=${collectionVariables.get('clientId')}
+      &client_secret=${collectionVariables.get('clientSecret')}`
+};
+
+const bearerToken = collectionVariables.get('bearerToken');
+const tokenExpirationDate = collectionVariables.get('tokenExpirationDate');
+if (pm.environment.name === 'LOCAL') {
+    const token = btoa(
+      `${collectionVariables.get('username')}:${collectionVariables.get('password')}`);
+    pm.request.headers.add(new Header(`Authorization: Basic ${token}`));
+} else if (bearerToken && tokenExpirationDate && tokenExpirationDate > new Date().getTime()) {
```

Comment:
I'd also add a condition to make sure the environment is either `DEV`, `TEST` or `PROD` in case people have other environments besides these four.
Think about it more, I'd recommend naming these environments more descriptive, e.g `OSU DEV`, `OSU TEST` and `OSU PROD` to avoid the confusion.

Reply:
I'd really like to find a better way to handle this. I was thinking of adding an `authType` variable to each environment and switching based on that instead of the environment name. What do you think?